hack/cluster-sync-handler.sh

```diff
@@ -8,21 +8,38 @@ ${KUBECTL} delete --ignore-not-found -f ${local_handler_manifest}
 # Set debug verbosity level for logs when using cluster-sync
 sed "s#--v=production#--v=debug#" ${local_handler_manifest} | ${KUBECTL} create -f -
 
-for i in {300..0}; do
-    # We have to re-check desired number, sometimes takes some time to be filled in
-    desiredNumberScheduled=$(${KUBECTL} get daemonset -n nmstate nmstate-handler -o=jsonpath='{.status.desiredNumberScheduled}')
+function getDesiredNumberScheduled {
+        echo $(${KUBECTL} get daemonset -n nmstate $1 -o=jsonpath='{.status.desiredNumberScheduled}')
+}
+
+function getNumberAvailable {
+        echo $(${KUBECTL} get daemonset -n nmstate $1 -o=jsonpath='{.status.numberAvailable}')
+}
+
+function isOk {
+        desiredNumberScheduled=$(getDesiredNumberScheduled $1)
+        numberAvailable=$(getNumberAvailable $1)
 
-    numberAvailable=$(${KUBECTL} get daemonset -n nmstate nmstate-handler -o=jsonpath='{.status.numberAvailable}')
 
-    if [ "$desiredNumberScheduled" == "$numberAvailable" ]; then
-        echo "nmstate-handler DS is ready"
-        break
+        if [ "$desiredNumberScheduled" == "$numberAvailable" ]; then
+          echo "$1 DS is ready"
+          return 0
+        else
+          return 1
+        fi
+}
+
+for i in {300..0}; do
+    # We have to re-check desired number, sometimes takes some time to be filled in
+    if isOk nmstate-handler && isOk nmstate-handler-worker ]; then
+       break
     fi
 
     if [ $i -eq 0 ]; then
-        echo "nmstate-handler DS haven't turned ready within the given timeout"
+        echo "nmstate-handler or nmstate-handler-worker DS haven't turned ready within the given timeout"
```

Comment:
The failing "isOk" is going to show you wich one has fail, so we can put here
"kubernetes-nmstate hasn't turned ready within the given timeout"

Reply:
'isOk' is printing just successful results not failures. It is possible to understand what handler caused the failure since it won't have a success message. But I would leave this messages as is, it is clearer to understand what happened (in case nmstate-handler-worker caused the failure, having the suggested message, the user will have to go to the scrtipt code to understand what exactly didn't turn ready).